app/services/api/charts/cp_performance.rb

```diff
@@ -0,0 +1,43 @@
+module Api
+  module Charts
+    class CPPerformance
+      # Calculate companies stats grouped by CP alignment in multiple series.
+      #
+      # @return [Array<Hash>] chart data
+      # @example
+      #   [
+      #     {
+      #       name: 'Below 2',
+      #       data: [ ['Coal Mining', 52], ['Steel', 73] ]
+      #     },
+      #     {
+      #       name: 'Paris Pledges',
+      #       data: [ ['Coal Mining', 65], ['Steel', 26] ]
+      #     }
+      #   ]
+      def cp_performance_all_sectors_data
+        all_companies = Company.published.includes(:sector, :latest_cp_assessment)
+
+        cp_alignment_data = {}
+        all_companies.each do |company|
+          next if company.cp_alignment.nil?
```

Comment:
we could change `Company.published` to `Company.published.with_cp_aligment` and remove `next if ..` from here.

Reply:
Right

Reply:
hmm, actually how to build `with_cp_alignment` scope, this depends on latest_cp_assessment. Dunno, maybe I'm already too tired today.